Specs/Core/EllipsoidRhumbLineSpec.js

```diff
@@ -0,0 +1,581 @@
+        var rhumb = new EllipsoidRhumbLine();
+        rhumb.setEndPoints(start, end);
+        expect(start).toEqual(rhumb.start);
+        expect(end).toEqual(rhumb.end);
+    });
+
+    it('gets heading', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 3);
+        var start = new Cartographic(CesiumMath.PI_OVER_TWO, 0);
+        var end = new Cartographic(Math.PI, 0);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(CesiumMath.PI_OVER_TWO).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes heading not going over the pole', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var start = new Cartographic(0, 1.2);
+        var end = new Cartographic(Math.PI, 1.5);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+
+        expect(0.0).not.toEqual(rhumb.heading);
+    });
+
+    it('computes heading going over the pole', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var start = new Cartographic(1.3, CesiumMath.PI_OVER_TWO);
+        var end = new Cartographic(0.0, CesiumMath.PI / 2.4);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+
+        expect(0.0).not.toEqual(rhumb.heading);
+    });
+
+    it('heading works when going around the world at constant latitude', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var start = new Cartographic(0.0, 0.3);
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, 0.3);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+
+        expect(CesiumMath.PI_OVER_TWO).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+
+        start = new Cartographic(3 * CesiumMath.PI_OVER_TWO, 0.3);
+        end = new Cartographic(CesiumMath.PI, 0.3);
+        var rhumb2 = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(-CesiumMath.PI_OVER_TWO).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes heading for vertical lines', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var start = new Cartographic(0.0, 1.2);
+        var end = new Cartographic(0.0, 1.5);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(0.0).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+
+        var rhumb2 = new EllipsoidRhumbLine(end, start, ellipsoid);
+        expect(CesiumMath.PI).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes no distance', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 3);
+        var start = new Cartographic(CesiumMath.PI_OVER_TWO, 0.0);
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, 0.0);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(0).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(0).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
```

Comment:
Is interpolation supposed to work here too? It's a little confusing since there's also 'throws without unique position' above, but these also seem to be non-unique points, but there's a valid surface distance so it might seem like interpolation should work.